pkg/types/gcp/machinepools.go

```diff
@@ -56,4 +61,83 @@ func (a *MachinePool) Set(required *MachinePool) {
 	if required.OSDisk.DiskType != "" {
 		a.OSDisk.DiskType = required.OSDisk.DiskType
 	}
+
+	a.EncryptionKey.Set(required.EncryptionKey)
+}
+
+// EncryptionKeyReference describes the encryptionKey to use for a disk's encryption.
+type EncryptionKeyReference struct {
+	// KMSKey is a reference to a KMS Key to use for the encryption.
+	//
+	// +optional
+	KMSKey *KMSKeyReference `json:"kmsKey,omitempty"`
+
+	// KMSKeyServiceAccount is the service account being used for the
+	// encryption request for the given KMS key. If absent, the Compute
+	// Engine default service account is used.
+	// See https://cloud.google.com/compute/docs/access/service-accounts#compute_engine_service_account
+	// for details on the default service account.
+	//
+	// +optional
+	KMSKeyServiceAccount string `json:"kmsKeyServiceAccount,omitempty"`
+}
+
+// Set sets the values from `required` to `e`.
+func (e *EncryptionKeyReference) Set(required *EncryptionKeyReference) {
+	if required == nil {
+		return
+	}
+	if e == nil {
+		e = &EncryptionKeyReference{}
```

Comment:
This will not change the function receiver at the caller location. Everything done to `e` will be lost.

Reply:
Hmm yeah good point, I've made it so these set functions return the var, do you think that's better or should I just be inlining this in the MachinePool Set?

Reply:
What would get us closest to the existing pattern would be something like the following in the MachinePool `Set` function.
```
if required.EncryptionKey != nil {
  if a.EncryptionKey == nil {
    a.EncryptionKey = &EncryptionKeyReference{}
  }
  a.EncryptionKey.Set(required.EncryptionKey)
}
```
And the EncryptionKeyReference `Set` function does not have to check for any nils.

Reply:
Updated, left the nil check in that matches the MachinePool just in case this ends up being called from somewhere else than what we are implementing here